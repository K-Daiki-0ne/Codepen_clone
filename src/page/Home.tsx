import React, { useState, useEffect } from 'react';
import { useLocalStorage } from '../hook/useLocalStorage';

export const Home: React.FC = (): JSX.Element => {
  const [html, setHtml] = useLocalStorage('html', '');
  const [css, setCss] = useLocalStorage('css', '');
  const [js, setJs] = useLocalStorage('js', '');
  const [srcDoc, setSrcDoc] = useState<string>('');

  useEffect(() => {
    const timeout = setTimeout(() => {
      setSrcDoc(`
        <html>
          <body>${html}</body>
          <style>${css}</style>
          <script>${js}</script>
        </html>
      `)
    }, 250)

    return () => clearTimeout(timeout)
  }, [html, css, js])

  return (
    <div className='pane top-pane'>

    </div>
  )
}