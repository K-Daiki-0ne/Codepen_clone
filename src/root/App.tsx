import React from 'react';
import { Home } from '../page/Home';
import './App.css';

const App: React.FC = (): JSX.Element => {
  return (
    <div className="App">
      <Home />
    </div>
  );
}

export default App;
