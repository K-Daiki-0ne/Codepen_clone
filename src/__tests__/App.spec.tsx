import React from 'react';
import { shallow } from 'enzyme';
import App from '../root/App';
import { Home } from '../page/Home';

describe('App.tsx ', () => {
  const wrapper = shallow(<App />);

  test('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('rendering test', () => {
    const wrap = wrapper.find('div');
    test('root', () => expect(wrap).toBeTruthy());
    test('children component', () => expect(wrap.find(Home)).toBeTruthy());
  })
})