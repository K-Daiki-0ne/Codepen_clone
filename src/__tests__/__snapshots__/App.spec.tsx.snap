// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `ShallowWrapper {}`;

exports[`App.tsx  snapshot test 1`] = `ShallowWrapper {}`;
